daemon/daemon.go

```diff
@@ -1189,6 +1190,9 @@ func NewDaemon() (*Daemon, *endpointRestoreState, error) {
 	// or IPv4 alloc prefix, respectively, retrieved by k8s node annotations.
 	log.Info("Initializing node addressing")
 
+	// Inject kvstore dependency into node pkg.
+	node.NodeReg = &nodeStore.NodeRegistrar{}
```

Comment:
I'm sensing that we will have a couple of those injections, can't this be in a `init()` function?

Reply:
Where would this init function be? I have to inject the dependency on the daemon side. I could factor out the initialization of node addressing into a helper function instead of having all of the code within `NewDaemon` in a future upload. I think that should be done regardless, as our bootstrapping functions are unwieldy.

Reply:
`init()` in the `daemon/daemon.go`

Reply:
This is a stylistic concern which I disagree with; I think we should collocate the bootstrapping code with the dependency injection as I have done here. It clearly indicates that we inject the dependency as part of bootstrapping that dependency as opposed to in a separate function. We can then factor out the bootstrapping from `NewDaemon` into separate functions that are done for separate subsystems.